### ISSUE TITLE
Blazor WASM (.NET6) does not allow AAD Authentication when Deployed to Azure Web app

### ISSUE THREAD

**Issue by igalfsg** [dotnet/aspnetcore#37206]
### Describe the bug
When deployed to Azure Web app with the preview version of .NET 6, the login button stops working and automatically redirects you to YOURSITE/authentication/login-failed. I looked in the logs and browser debug tools and couldn't see an error occurring anywhere. When running the code in localhost there is no problem.  
 
![image](https://user-images.githubusercontent.com/8607853/135700120-ce718499-9b89-48f6-96a7-90121214f5d9.png)

### To Reproduce
You can repro de issue live by going to https://sshmantest.azurewebsites.net/ and clicking login in the top right.

I have also created a repo with code and it includes the GitHub Action I am using to deploy the code to Azure:
https://github.com/coding-flamingo/BlazorWasmWithAADAuth

The Azure environment is an Azure Web App in a Linux app plan.
App plan:
![image](https://user-images.githubusercontent.com/8607853/135700187-db6bacff-7ab2-4980-b396-182a164e11a2.png)

Web App Configuration
![image](https://user-images.githubusercontent.com/8607853/135700205-fe305192-9765-4cc2-ac8b-90dc4bc9db23.png)

### Exceptions (if any)
No Exceptions were found

### Further technical details
- ASP.NET Core version .NET 6

EDIT: To prove that it works on .net 5 five I moved the .NET 6 app to https://testnet2.azurewebsites.net/  and the working .NET 5 is now in https://sshmantest.azurewebsites.net/

**Comment by igalfsg** [comment]
![image](https://user-images.githubusercontent.com/8607853/137602842-e182fcc6-13aa-4bb4-9169-3a5af042766a.png)
@knight1219 this are my settings for the client side. They should be fine this was a working application before the change to .NET 6

**Comment by igalfsg** [comment]
@javiercn my issue is not for configuring it wrongly.. this app works on my dev machine.. and if I downgrade to ,NET 5 it works on Azure on this same web app..

**Comment by igalfsg** [comment]
@javiercn my app is not misconfigured, it literally works on my machine with the same app ID and if I downgrade to .NET5 it works in Azure. As mentioned in my previous comments this happens before going to the STS so it doesn't seem like an Azure AD issue. Seems like an important issue to solve before people try to use it in production. more that it works on debug

**Comment by knight1219** [comment]
![image](https://user-images.githubusercontent.com/6771697/140835272-eae92148-c5cd-4b27-8e7e-695a92adf7a8.png)

I just tried your demo site and I'm getting this back in the console log. Are you seeing this too?

**Comment by igalfsg** [comment]
I see those errors when it works in local machine as well so I assumed that it was  Blazor looking for an authenticated user before authentication and were safe to ignore 
![image](https://user-images.githubusercontent.com/8607853/140940235-3095f907-e08b-455f-b1c0-f8bc85122425.png)

**Comment by igalfsg** [comment]
and if I literally just revert back to .NET 5 it works 
![image](https://user-images.githubusercontent.com/8607853/140941898-77eff325-36b8-4f2b-ab49-698222e4e956.png)
here is a link to the commit reverting https://github.com/coding-flamingo/BlazorWasmWithAADAuth/commit/f3d16ccb2c13cab6113706944ecb8caf903f05cc
and here are the new app settings:
![image](https://user-images.githubusercontent.com/8607853/140942143-cc30f0fb-ed11-4f9e-b342-90c91e9c9fff.png)

**Comment by igalfsg** [comment]
Update this can be "Fixed" by adding the following to the Client:
```
<ItemGroup>
<TrimmerRootAssembly Include="Microsoft.Authentication.WebAssembly.Msal" />
</ItemGroup>
```
I will be removing my Azure test environments, but will leave this issue open since this is just a workaround until Microsoft fixes the issue...

thanks to u/RoderoDaVinci for finding this fix.

**Comment by javiercn** [comment]
We believe this has been addressed in 7.0 as part of https://github.com/dotnet/aspnetcore/pull/43954.

The fix should be available in RC2.
